src/core/MenuSelect/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { InputAdornment } from "@material-ui/core";
+import {
+  AutocompleteProps,
+  AutocompleteRenderInputParams,
+  AutocompleteRenderOptionState,
+} from "@material-ui/lab";
+import React from "react";
+import {
+  InputBaseWrapper,
+  StyledAutocomplete,
+  StyledInputBase,
+  StyledMenuItem,
+  StyledSearchIcon,
+  StyleProps,
+} from "./style";
+
+// (thuang): This requires option to have a `name` property.
```

Comment:
I'm not sure if it's too restrictive to demand options to **always** have a `name` property. But we need this guarantee in order to provide other default utilities. E.g., `defaultGetOptionLabel()` and `defaultRenderOption()`